spec/fixtures/mock_upstream.lua

```diff
@@ -105,7 +143,23 @@ local function parse_post_data(headers)
     end
   end
 
-  return data, form, params
+  return {
+    args    = ngx.req.get_uri_args(),
+    data    = data,
+    form    = form,
+    headers = headers,
+    params  = params,
+    url     = string.format("%s://%s%s", ngx.var.scheme,
+                            ngx.var.host, ngx.var.request_uri),
```

Comment:
Considering our test instance runs on a non-default HTTP port, I wonder if it would make sense to include the port in the URL? Usually the port should be included in URL nomenclature and may only be omitted if it is a default one iirc (`80`/`443`), no?

Doing so would also allow us to use `mock_upstream_url` in our tests, instead of `mock_upstream_scheme .. "://" .. mock_upstream_host`.